blogposts/2022/release-3.39

```diff
@@ -0,0 +1,75 @@
+    category: Search
+  - description: "Built-in authentication (i.e. username and password) now supports account lockout after consecutive failed sign-in attempts. New config options have been added under `\auth.lockout\` to customize the threshold for failed attempts and length of lockouts."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33999
+    category: Admin
+---
+
+Sourcegraph 3.39 is now available! In this release:
+
+## Onboard developers fast with Notebooks, now Generally Available
+
+Built with onboarding in mind, Notebooks make documentation effortless, stable, and fun. Notebooks are inspired by the Jupyter Notebooks project and feature four unique block types, an intelligent notepad for creating notebooks on the fly, and deep integration with Sourcegraph's code search so you can always find the code you need.
+
+<video class="blog-image" title="Notebooks overview" alt="An example of using a notebook." loop autoplay muted playsinline>
+  <source src="https://user-images.githubusercontent.com/1657213/163855878-4355bcb0-5407-4307-b3f2-4581e1ab2fe2.mp4">
+</video>
+
+Web-based notebooks are the easiest way to create and share notebooks with your team or company, but we also support file-based notebooks via a special `.snb.md` file type. With file-based Notebooks, you can transform your existing documentation and augment it with Sourcegraph query blocks. You can also embed notebooks on any web page, enabling easy integration with your own knowledge management systems, so you can meet your developers where they are.
+
+Notebooks are now available for all users in Sourcegraph 3.39, and you can also explore some of our [public notebooks on Sourcegraph Cloud](https://sourcegraph.com/notebooks?tab=explore) with no account required.
+
+To learn more about Notebooks, [read the docs](https://docs.sourcegraph.com/notebooks).
+
+## Initial support for filtering code insights using search contexts
+
+<div style="text-align:center">
+<img class="blog-image" title="Code insights have search context filters" alt="Image of a search context being used to filter a code insight" src="https://storage.googleapis.com/sourcegraph-assets/blog/3.39/code_insights_3.39_dynamic_context_filters.png
+">
+</div>
+
+Using a search context lets you reuse the same `repo:` pattern to filter multiple insights to a subset of your repos, and then maintain and update that group of repos in just one place, the search context. Your filtered insights will automatically reflect the current state of any updated context. Using a context is also a more succinct way to reference an extended `repo:` filter for reuse as a filter.
+In this initial release, you can use the `repo:` field of [query-based search contexts](https://docs.sourcegraph.com/code_search/how-to/search_contexts#beta-query-based-search-contexts) to filter your code insights. We're continuing to explore enabling other capabilities of search contexts to use as filters.
```

Comment:
Waiting for a docs link for this one (@Joelkw )

Reply:
Docs link added! It'll be live once this is live: https://github.com/sourcegraph/sourcegraph/pull/34162 , but because it's an anchor it's safe to publish around the same time (still takes you to the correct page) 
```suggestion
In this initial release, you can use the `repo:` field of [query-based search contexts](https://docs.sourcegraph.com/code_search/how-to/search_contexts#beta-query-based-search-contexts) to filter your code insights. We're continuing to explore enabling other capabilities of search contexts to use as filters. To learn more, [read the docs](https://docs.sourcegraph.com/code_insights/explanations/code_insights_filters#context-query-based-search-context-filters). 
```